src/osdatahub/filters.py

```diff
@@ -1,21 +1,79 @@
-from typing import Callable, Union
-from enum import Enum
+from typing import Union
+
 from osdatahub import Extent
 
 
-def spatial_filter(operator: str, extent: Extent) -> str:
+class Filter:
+    def __init__(self, xml: str):
+        self.xml: str = xml
+
+    @staticmethod
+    def _apply_op(filter1: str, filter2: str, operation: str) -> str:
+        return f"<ogc:{operation}>{filter1}{filter2}</ogc:{operation}>"
+
+    def __and__(self, other) -> "Filter":
+        return Filter(self._apply_op(self.xml, other.xml, "And"))
+
+    def __bool__(self):
+        raise NotImplementedError("Did you use a boolean operation, meaning to use a bitwise operation instead?")
+
+    def __iadd__(self, other) -> "Filter":
+        self.xml = self._apply_op(self.xml, other.xml, "And")
+        return self
+
+    def __ior__(self, other) -> "Filter":
+        self.xml = self._apply_op(self.xml, other.xml, "Or")
+        return self
+
+    def __or__(self, other) -> "Filter":
+        return Filter(self._apply_op(self.xml, other.xml, "Or"))
+
+    def __repr__(self) -> str:
+        return f"<Filter object with xml {self.xml!r}>"
+
+    def __str__(self) -> str:
+        return self.xml
+
+
+def filter_or(filter1: Filter, filter2: Filter) -> Filter:
+    """Constructs an OGC XML filter that performs an 'or' on the two give filters
+
+    Args:
+        filter1 (Filter):
+        filter2 (Filter):
+
+    Returns:
+        Filter: A valid OGC XML filter
+    """
+    return filter1 | filter2
+
+
+def filter_and(filter1: Filter, filter2: Filter) -> Filter:
```

Comment:
I think this could take an arbitrary number of filters, and perform an and on all of them. Same with filter_or()